packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3614,6 +3627,73 @@ export class TransactionController extends BaseController<
     return transactionMeta;
   }
 
+  async #updateFirstInteraction(
+    transactionMeta: TransactionMeta,
+    {
+      traceContext,
+    }: {
+      traceContext?: TraceContext;
+    } = {},
+  ) {
+    const {
+      chainId,
+      id: transactionId,
+      txParams: { to, from },
+    } = transactionMeta;
+
+    const request: FirstTimeInteractionRequest = {
+      chainId,
+      to,
+      from,
+    };
+
+    let firstTimeInteractionResponse: FirstTimeInteractionResponse;
+
+    try {
+      validateFirstTimeInteraction(request);
```

Comment:
The params are validated when we first add the transaction in `addTransaction`, so why do we need to validate an object generated internally?

Reply:
You're right, I missed that `addTransaction` is also validating. However I feel like we still need it because currently this feature tied to `txParams.to` property, so we need this validation still.